### PR TITLE
use imagemagick and libraw for raw image support

### LINKED ISSUE #2156: [BUG] Sony RAW (.ARW) unsupported
### The bug

I found https://github.com/immich-app/immich/issues/1109, but do not believe the resolution is fair. The Sony RAW format is very popular and is a major blocker for adoption.

The linked [libvips issue](https://github.com/libvips/libvips/issues/1304) suggests there is support for the file format, but `magickload` must be used instead.

### The OS that Immich Server is running on

N/A

### Version of Immich Server

v1.52.1

### Version of Immich Mobile App

N/A

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
N/A
```

### Additional information

N/A

### PR DESCRIPTION
**Pull request by uhthomas**
imagemagick and libraw have generally good support for raw images, including
Sony's ARW format. These tools should also allow Immich to support many more
image formats in future without any major code changes.

https://www.libraw.org/supported-cameras

I've tested and verified this change with .ARW files and other standard formats.

Fixes: https://github.com/immich-app/immich/issues/2156

### CODE REVIEW

**Comment by uhthomas** [comment]
I've tested this, and am awaiting a response from the sharp author regarding https://github.com/lovell/sharp/issues/3612#issuecomment-1576714726. Please don't merge yet.

**Comment by samip5** [comment]
I think the idea was to use darktable.

See https://github.com/immich-app/immich/pull/2198

**Comment by uhthomas** [comment]
> I think the idea was to use darktable.
> 
> See #2198

Also see https://github.com/immich-app/immich/issues/2156#issuecomment-1561955606.

**Comment by uhthomas** [comment]
It would be nice to merge https://github.com/immich-app/immich/pull/2736 before this.

File: server/Dockerfile
Comment (uhthomas) [inline server/Dockerfile:26]: `-dev` packages may not be needed here.

Comment (uhthomas) [reply server/Dockerfile:26]: I tested without `vips-dev` and it wasn't happy at all. I presume all it wants is `vips-cpp`.

Comment (uhthomas) [reply server/Dockerfile:26]: Nope, not happy with `vips-cpp` either. I'll leave the `-dev` packages as is :)

File: server/src/domain/media/media.service.ts
Comment (uhthomas) [inline server/src/domain/media/media.service.ts:65]: Q for reviewers: Should we just remove this? vips/imagemagick should do this automatically. Might be nice to reduce code footprint and complexity. What do you think?

Comment (jrasm91) [reply server/src/domain/media/media.service.ts:65]: This was a work around (fallback) for raw images that had an embedded thumbnail. If raw photos work without this I think we can definitely remove it.

Comment (uhthomas) [reply server/src/domain/media/media.service.ts:65]: https://github.com/immich-app/immich/issues/2744

**Review by jrasm91 — COMMENTED** [review]
How big of an image size jump does moving away from alpine cause?

**Comment by uhthomas** [comment]
> How big of an image size jump does moving away from alpine cause?

Hm? We are using alpine.

File: web/src/lib/utils/asset-utils.ts
Comment (uhthomas) [reply web/src/lib/utils/asset-utils.ts:146]: You're right, I'm not sure how I did that. Thank you.

**Comment by jrasm91** [comment]
> > How big of an image size jump does moving away from alpine cause?
> 
> Hm? We are using alpine.

Ah, sorry I mis-read the change. `alpine` got bumped to the next line with the addition of the hash.

File: web/src/lib/utils/asset-utils.ts
Comment (uhthomas) [reply web/src/lib/utils/asset-utils.ts:146]: @jrasm91 I did one even better and added a test for this. The test also revealed that null or undefined values were not handled correctly, and has been fixed.

**Review by elliotclee — COMMENTED** [review]
Hi,

Thanks for working on this. I like the way you changed to using arrays instead of endless if() statements in a number of areas.

I think it would make sense to add support for all the filetypes that libraw can handle. I came looking for .cr2 (image/x-canon-cr2), and I gather there are several other types that it would be helpful to accept. See https://invent.kde.org/graphics/digikam/-/blob/master/core/libs/rawengine/drawfiles.cpp for a fairly complete list of the file types supported by libraw, as well as https://github.com/angryziber/gnome-raw-thumbnailer/blob/master/data/raw-thumbnailer.xml for info on the MIME types for them.

Hope this helps,

**Comment by uhthomas** [comment]
> Hi,
> 
> Thanks for working on this. I like the way you changed to using arrays instead of endless if() statements in a number of areas.
> 
> I think it would make sense to add support for all the filetypes that libraw can handle. I came looking for .cr2 (image/x-canon-cr2), and I gather there are several other types that it would be helpful to accept. See https://invent.kde.org/graphics/digikam/-/blob/master/core/libs/rawengine/drawfiles.cpp for a fairly complete list of the file types supported by libraw, as well as https://github.com/angryziber/gnome-raw-thumbnailer/blob/master/data/raw-thumbnailer.xml for info on the MIME types for them.
> 
> Hope this helps,

I would love to do this, but I am afraid for the scope of this PR becoming too large, which is why I separated some changes into #2736. I think the best thing to do would be to merge this as-is, closing the original Sony ARW issue and create another issue like #2743 to support the remaining libraw formats. Does that sound good to you? I can create that issue now.

**Comment by jrasm91** [comment]
I think it is OK to add, test, and verify support for specific formats. With the change to the base image it'll be pretty trivial to add support for additional formats. Whether we do all that testing in this PR or it follow-up PRs seems less important. I don't think @uhthomas needs to commit to testing every single format type before we merge this support for ARW.

**Comment by uhthomas** [comment]
I've created #2745 to track the implementation of all libraw formats.

**Review by jrasm91 — APPROVED** [review]

**Comment by uhthomas** [comment]
It looks like this job was cancelled and is confusing GitHub. Any idea what happened, or if it's intentional?

https://github.com/immich-app/immich/actions/runs/5248322532/jobs/9479643740?pr=2668

**Comment by uhthomas** [comment]
Strange, I made force pushed and now a different build was 'cancelled' https://github.com/immich-app/immich/actions/runs/5248561828/jobs/9480220088?pr=2668.

**Comment by jrasm91** [comment]
I have had some issues of late with force-pushing having weird impact on the running jobs.

**Comment by uhthomas** [comment]
Ah, https://www.githubstatus.com/incidents/csnbwc85chp2.

File: web/src/lib/utils/asset-utils.ts
Comment (uhthomas) [inline web/src/lib/utils/asset-utils.ts:130]: I don't think that will work?

**Comment by uhthomas** [comment]
It seems like this image is missing quite a few runtime dependencies. Namely imagemagick-dev and vips-magick.

https://github.com/immich-app/immich/blob/af86712df3b093cc2a342cff3279833ba055d520/server/Dockerfile#L26

**Pull request merged**
